cub/test/catch2_test_block_load.cu

```diff
@@ -33,37 +33,28 @@
 #include <c2h/catch2_test_helper.h>
 
 template <int ItemsPerThread, int ThreadsInBlock, cub::BlockLoadAlgorithm LoadAlgorithm>
-struct output_idx
+static __device__ int get_output_idx(int item)
 {
-  static __device__ int get(int item)
-  {
-    return static_cast<int>(threadIdx.x) * ItemsPerThread + item;
-  }
-};
-
-template <int ItemsPerThread, int ThreadsInBlock>
-struct output_idx<ItemsPerThread, ThreadsInBlock, cub::BlockLoadAlgorithm::BLOCK_LOAD_STRIPED>
-{
-  static __device__ int get(int item)
+  if (LoadAlgorithm == cub::BlockLoadAlgorithm::BLOCK_LOAD_STRIPED)
```

Comment:
`if constexpr` when?

No change requested until we require C++17

Reply:
I had `_CCCL_IF_CONSTEXPR` initially, but then you have to put braces around both branches and add an unreachable in the dead code for nvcc. Too much LOCs for what this function tries to achieve.

Reply:
sad but true